### ISSUE TITLE
Unify/simplify the config file search across OS versions. ALT: Allow all versions to honor XDG_CONFIG_HOME if set

### ISSUE THREAD

**Issue by eldorel** [vim/vim#14786]
I use multiple operating systems on a daily basis. 
At the moment, I have to maintain several different config file layouts/symlinks/etc for each OS due to the inconsistent location assumptions across VIM/GVIM versions. 

In my opinion, having separate logic and file locations for each OS is unnecessary and introduces unneeded complexity to the end user experience.  
Even the ( frankly silly ) option of having all versions look at all legacy locations would simplify the end user experience. 

The new patch allowing the linux/Unix versions to use the XDG variables has the potential to be a huge step toward simplifying that, but Windows (and possibly MacOS) versions do not honor that variable.

**Describe the solution you'd like**

I would like to see the logic for locating the vim config files made consistent across all versions, so that users can simply pick one location and use it across all of their systems. 

$HOME/.vimrc; $HOME/.config/vim/vimrc; etc are all completely valid file locations on all major operating systems and fileystems... 

The historical reasons for requiring completely different filenames for the Windows version is no longer present, and modern versions of NTFS/FAT have absolutely no issues with periods in a filename. 

Additionally, the myriad of compatibility improvements such as terminal emulation, powershell, compatibility layers such as WSL or cygwin, etc have introduced the ability to have multiple versions of vim in sub-environments on a single device...  

End users should not be required to manage symlinks and/or multiple config files in order to have a consistent configuration.

**Additional Note**

Personally, I just want the windows and MacOS versions to check for the XDG environment variables in exactly the same manner that the current linux build does.  
This would allow me to configure a single variable when I initially setup a new system, and simplify the process of maintaining a syncronized vimrc across all of my devices.

However, I feel that rather than implementing a specific fix to allow power users to have the benefit of a consistent configuration, the entire system for handling config file locations needs to be overhauled in order to allow *all* users to have a more consistent and streamlined experience.

**Comment by saccarosium** [comment]
XDG environment variables are a Linux spefic. Although on MacOS in the terminal they are use, due to be UNIX . On windows there are other sets of guidelines that should be followed.

**Comment by eldorel** [comment]
> XDG environment variables are a Linux spefic. Although on MacOS in the terminal they are use, due to be UNIX . On windows there are other sets of guidelines that should be followed.

Environment variables are trivial to add on all OS's. 

Part of why I made the request is because I am using the XDG variables on several windows 10 and 11 systems to configure other cross-platform software, and it was delightfully easy to do. 

Then I installed vim on my new laptop, and had to think about where I needed to place all of the different config files so that it works the same way no matter what terminal/environment I execute from. 

On the plus side, I found out that setting XDG_CONFIG_HOME in the windows user environment also makes that variable visible from WSL/Cygwin/Wezterm/Powershell/etc... 

So now I just have a handful of identical single line .vimrc/_vimrc files with a single 'source' directive that points to the _actual_ config file.  

`source $XDG_CONFIG_HOME\vim\vimrc
`

**Comment by jlanzarotta** [comment]
Yes, this would be fantastic if this worked under Windows.  There are a lot of other "Windows" application that are starting to use the .config directory, and it would be nice if Vim followed suit.  Having a centralized configuration directory is very handy for knowing and backing up all your configuration files.

**Comment by saccarosium** [comment]
I think they use `.config` because either they don't know the appropriate location or simply they do not care about Windows and use the same location that they use on UNIX. 

I do not use windows but sounds wrong. At least [looking at the Windows documentation](https://learn.microsoft.com/en-us/windows/win32/shell/knownfolderid).

Also looking at libraries that helps you locating files in the appropriate spot:
- for golang: https://github.com/adrg/xdg doesn't use `.config`
- for rust: https://docs.rs/etcetera/latest/etcetera/ doesn't use `.config`
- etc
